packages/ui/src/components/Tour/tourContext.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useContext, useState } from 'react'
+
+type TourContextValue = {
+    currentStep?: number
```

Comment:
Why everything is optional?

Reply:
You should provide proper default value instead

Reply:
Context if basically a 3 things (types + defaults + actual values)

Reply:
So you can solve it like this:

1. Types. You should say value is defined
```typescript
type TourContextType = {
    currentStep: number
    setCurrentStep: React.Dispatch<React.SetStateAction<number>>
}
```
2. Defaults. What to return if no Provider not found in parent components
```typescript
const TourContext = React.createContext<TourContextValue>({
    currentStep: 0
    setCurrentStep: () => ({})
})
```
So, now if you decided to use context without a provider, you will get this defaults

3. Actual values
```
const TourContextProvider = ...
```